torch/_dynamo/symbolic_convert.py

```diff
@@ -1086,10 +1086,23 @@ def run(self):
             except TensorifyScalarRestartAnalysis:
                 raise
             except BackendCompilerFailed:
+                raise
+            except RuntimeError as e:
+                if hasattr(e, "msg") and "Could not guard on data-dependent" in e.msg:
```

Comment:
I think my main question is... do we really only want to log the graph when we see data-dependent errors. It seems like a generally useful whenever we blow up in the middle of tracing.

I guess the main risk is that we risk spamming users with larger errors messages, which is counter to @williamwen42 's work to make "internal compiler errors" show up more nicely to users.

@williamwen42 how are you thinking about balancing "make internal compiler errors less scary to users" with "keep enough info in them to help devs debug? We might want a config/envvar to toggle verbose stacktraces throughout PT2?

Reply:
Is "Could not guard on data-dependent" an internal compiler error or is it user error?

Reply:
I'll put up a separate PR that makes this show for all errors so we can do partial reverts.

Reply:
@zou3519 it's a UserError

```
type(e)
<class 'torch._dynamo.exc.UserError'>
type(e).mro()
[<class 'torch._dynamo.exc.UserError'>, <class 'torch._dynamo.exc.Unsupported'>, <class 'torch._dynamo.exc.TorchDynamoException'>, <class 'RuntimeError'>, <class 'Exception'>, <class 'BaseException'>, <class 'object'>]
```

Reply:
hmm (I think?) Richard was asking more broadly whether the data dependent error is an example of a user error (that they can fix with model tweaks) or a compile-side error (that we need to fix by improving the compiler error).

If feels to me like data-dependent errors kind of blur the line between the two... would you agree with that? (some of the time the user can fix their error with more `torch._check()` calls, other times it turns our that we need to beef up our dynamic shapes infra to help them)

Reply:
We can log to an existing logging artifact instead of using global `print` to implement toggle-able stack traces for data-dependent errors today.